### PR TITLE
some basic ui stuff

### PR DESCRIPTION
**Pull request by anranlee99**
pls fix merge

### CODE REVIEW

**Comment by anranlee99** [comment]
no conflict now?

File: src/main.ts [hunk 1]
```diff
@@ -3,25 +3,82 @@ import * as Tone from 'tone';
...
+                    Tone.Transport.stop()
+                    Tone.Transport.start()
+                    if(i == 3){
+                        this.synth.triggerAttackRelease("C5", "8n")
+                    } else {
+                        this.synth.triggerAttackRelease("C4", "8n")
+                    }
+                    i++
+                },
+                repeat: 3
```
Comment (s5bug) [inline src/main.ts]: Essentially what you should be doing is `sync`ing all the synths to the `Transport`, and then only changing them when the underlying music changes. Essentially, what you have on the `Transport` is data-driven, and then all the inputs control the `Transport`.

```ts
// one time
this.synth.sync() // sync the synth to the transport
this.synth.triggerAttackRelease("C4", "8n", 0)
this.synth.triggerAttackRelease("C4, "8n", "2n")
this.synth.triggerAttackRelease("C4", "8n", "1n")
```
```ts
// every click
Tone.Transport.stop()
Tone.Transport.start()
```

File: src/main.ts [hunk 2]
```diff
@@ -33,6 +90,13 @@ let game = new Phaser.Game({
         width: 1920,
         height: 1080
     },
+    physics: {
+        default: 'arcade',
+        arcade: {
+            debug: true,
+        }
+    },
```
Comment (s5bug) [inline src/main.ts]: Is physics used here at all?

**Review by s5bug — CHANGES_REQUESTED** [review]

File: src/main.ts [hunk 3]
```diff
@@ -3,25 +3,91 @@ import * as Tone from 'tone';
...
+                callback: () => {
+                    this.circles[i].setAlpha(0.5)
+
+                    this.synth.sync()
+                    if (i == 3) {
+                        this.synth.triggerAttackRelease("C5", "8n")
+                    } else {
+                        this.synth.triggerAttackRelease("C4", "8n")
+                    }
+                    i++
```
Comment (s5bug) [inline src/main.ts]: This is still incorrect:

`sync` and `triggerAttackRelease` should only be done once: this puts the notes on the Transport (think of the Transport as a piece of sheet music)

`Transport.start()` and `Transport.stop()` control the musician playing the sheet music, and _these_ are what you should be calling repeatedly

**Review by s5bug — CHANGES_REQUESTED** [review]

File: src/main.ts [hunk 4]
```diff
@@ -3,25 +3,89 @@ import * as Tone from 'tone';
...
             Tone.Transport.stop()
-            Tone.Transport.start()
+            Tone.Transport.bpm.value = 60;
+            const seq = new Tone.Sequence((time, note) => {
+                this.synth.triggerAttackRelease(note, 0.1, time);
+                // subdivisions are given as subarrays
+                this.circles[i].setAlpha(0.5)
+                i++
+            }, ["C4", "C4", "C4", "C5"], "4n").start(0, 0);
+            seq.loop = 1;
```
Comment (s5bug) [inline src/main.ts]: This should be moved to after the `sync` instruction as to not be done every time the mouse is pressed

File: src/main.ts [hunk 5]
```diff
@@ -3,25 +3,89 @@ import * as Tone from 'tone';
...
+        }).setOrigin(0.5).setPadding(10)
+        let mask = graphics.fillRoundedRect(0, 0, 200, 200, 32)//.setDepth(iconRect.depth + 1)
+        mask.copyPosition({
+            x: iconRect.x - iconRect.width / 2,
+            y: iconRect.y - iconRect.height / 2
+        })
+        iconRect.createGeometryMask(mask)
+        iconRect.setInteractive({ useHandCursor: true, draggable: true })
+        iconRect.on(Phaser.Input.Events.DRAG, (pointer: Phaser.Input.Pointer, dragX: number, dragY: number) => {
+
```
Comment (s5bug) [inline src/main.ts]: Either rename to `_pointer` or add a useless line `pointer;` to silence the Typescript warning

**Review by s5bug — CHANGES_REQUESTED** [review]

**Review by s5bug — APPROVED** [review]

**Pull request merged**